File: core/LargestTriangleArea/LargestTriangleArea.py
from typing import List


def largestTriangleArea(points: List[List[int]]) -> float:
    length, res = len(points), 0.
    for i in range(length - 2):
        for j in range(i + 1, length - 1):
            for k in range(j + 1, length):
                res = max(res, abs(points[i][0]*points[j][1] + points[j][0]*points[k][1] + points[k][0]*points[i][1] - points[i][0]*points[k][1] - points[j][0]*points[i][1] - points[k][0]*points[j][1]) / 2.0)

    return res

if __name__ == '__main__':
    print(largestTriangleArea([[1,0],[0,0],[0,1]]))

File: core/LargestTriangleArea/LargestTriangleArea.java
package com.dbc;

public class LargestTriangleArea {
    public double largestTriangleArea(int[][] points) {
        double res = 0;
        for (int i = 0; i < points.length - 2; i++) {
            for (int j = 0; j < points.length - 1; j++) {
                for (int k = 0; k < points.length; k++) {
                    float value = points[i][0]*points[j][1]+points[j][0]*points[k][1]+points[k][0]*points[i][1]-points[j][0]*points[i][1]-points[j][1]*points[k][0]-points[k][1]*points[i][0];
                    res = Math.max(res, Math.abs(value) / 2.);
                }
            }
        }
        return res;
    }
}
